### DISCUSSION TITLE
Access element innerText / innerHTML

### DISCUSSION THREAD

**Discussion by mraspe** [php-webdriver/php-webdriver#825] [Q&A] [answered]
### What are you trying to achieve? (Expected behavior)
Attribute "innerText/innerHTML" of an element (td) should not be empty (in Firefox).

### What do you get instead? (Actual behavior)
In Chrome I get the correct string of an element attribute (innerText). In Firefox the string is always empty.

### How could the issue be reproduced? (Steps to reproduce)
In a PHP test I try to get the content of an element (td):  $column->getAttribute('innerText');
With Selenium server version 3.8.1 it had still worked. With version 3.141.59 it only works in chrome.

### Details
* Php-webdriver version: 1.8.3
* PHP version: 7.2.34
* Selenium server version: 3.141.59
* Operating system: Win10
* Browser used + version: Firefox 81.0, Chrome 85.0.4183.121
* Geckodriver: 0.27.0
* Chrome Diver: 85.0.4183.38

**Comment by OndraM** [comment, marked as answer]
To get text value of an element, you can use native WebDriver method `$element->getText()`.
To get JavaScript property like `innerHTML` or `childElementCount` etc., **use `getDomProperty()` method** (see example below).
`getAttribute()` could be used only to get value of DOM element attribute (like "class", "id", "style" etc.) . 

```php
$innerHtml = $driver->findElement($element)->getDomProperty('innerHTML');
```
[See wiki](https://github.com/php-webdriver/php-webdriver/wiki/Example-command-reference#get-property-of-an-element)

Alternatively, you can ale use `executeScript()` to get the same result:
```php
$innerHtml = $driver->executeScript('return arguments[0].innerHtml', $element)
```

**Reply by rmontanana** [reply]
```php
$innerHtml = $driver->executeScript('return arguments[0].innerHtml', [$element])
````
actually works for me
Thx!

**Reply by OndraM** [reply]
Actually, there is already a better way! You can try:

```php
$innerHtml = $driver->findElement(WebDriverBy::id('myelement'))->getDomProperty('innerHTML');
```

See [wiki.](https://github.com/php-webdriver/php-webdriver/wiki/Example-command-reference#get-property-of-an-element)

I will update the answer to be up-to-date.

**Reply by rmontanana** [reply]
Indeed it does, thank you very much!

**Comment by mraspe** [comment]
Hi,
okay, thank you!

I was wondering why you get different values with "innerText" (empty, not empty) in Chrome and Firefox. With $element->getText it works,
